api/user/schema.py

```diff
@@ -112,11 +113,46 @@ def mutate(self, info, email, **kwargs):
         exact_user = query_user.filter(UserModel.email == email).first()
         if not exact_user:
             raise GraphQLError("User not found")
-        user_role = UsersRole.query.filter_by(user_id=exact_user.id).first()
-        new_role = kwargs.pop('role_id')
-        user_role.role_id = new_role
-        with SaveContextManager(user_role, 'Role id', new_role):
-            return ChangeUserRole(user=exact_user)
+
+        if not exact_user.roles:
+            raise GraphQLError('user has no role')
+
+        new_role = RoleModel.query.filter_by(id=kwargs['role_id']).first()
+        if not new_role:
+            raise GraphQLError('invalid role id')
+
+        exact_user.roles[0] = new_role
```

Comment:
@dnuwa any reason why here you update the user's role using the index while in the CreateUserRole mutation you use the append method? Try and be more consistent.

Reply:
`exact_user.roles` returns a list. so what I am trying to achieve here is to replace the value returned at index 0 with a new value in `new_role`. That's why the implementation here is different. This mutation takes care of an instance where the user needs to update a role.